packages/ra-core/src/routing/RestoreScrollPosition.tsx

```diff
@@ -0,0 +1,30 @@
+import { ReactNode } from 'react';
+import { useRestoreScrollPosition } from './useRestoreScrollPosition';
+
+/**
+ * A component that tracks the scroll position and restores it when the component mounts.
+ * @param children The content to render
+ * @param key The key under which to store the scroll position in the store
+ * @param debounceMs The debounce time in milliseconds
+ *
+ * @example
+ * import { ListBase, RestoreScrollPosition } from 'ra-core';
+ *
+ * const MyCustomList = (props) => (
+ *   <RestoreScrollPosition key="my-list>
+ *     <ListBase {...props} />
+ *   </RestoreScrollPosition>
+ * };
```

Comment:
I think you should use another example than a list, as it's useless in that case

Reply:
Done but somehow git does not show this file as outdated